src/doom/r_plane.c

```diff
@@ -484,7 +484,18 @@ void R_DrawPlanes (void)
 
 		if ((unsigned) dc_yl <= dc_yh) // [crispy] 32-bit integer math
 		{
-		    angle = ((an + xtoviewangle[x])^flip)>>ANGLETOSKYSHIFT;
+			// [crispy] Draw skies horizontally linear.
+			// Taken from GZDoom and refactored for integer math.
+			if (crispy->linearsky) 
+			{
+				angle_t xangle = ((viewwidth / 2 - x) * ((SCREENWIDTH<<6) / viewwidth))
+                                  * (ANG90 / (NONWIDEWIDTH<<6));
```

Comment:
OK, this should be good, as long as you're ok with the refactoring, and the Crispness menu layout.

Reply:
Out of curiosity, what happens if you regenerate the whole `xtoviewangle[]` array with this formula?

Reply:
If you mean like this: 

    if (crispy->linearsky) 
    {
        xtoviewangle[x] = ((viewwidth / 2 - x) * ((SCREENWIDTH<<6) / viewwidth))
            * (ANG90 / (NONWIDEWIDTH<<6));
    }
    angle = ((an + xtoviewangle[x])^flip)>>ANGLETOSKYSHIFT;

it gives a bunch of visual errors on some walls (but maybe I'm missing something?), plus `xtoviewangle[x]` would need regenerating when `crispy->linearsky` is disabled.

![DOOM0000](https://user-images.githubusercontent.com/47063032/84719530-983f9a80-af49-11ea-8a14-105e387abcf6.png)

Reply:
If you're looking to avoid calculating `xangle` in `R_DrawPlanes` every time, I could move it to a new array that's declared and calculated at the same time as `xtoviewangle` in `R_InitTextureMapping` in `r_main.c`? Does that get called less often?